junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -120,8 +335,13 @@ inline fun <reified T : Throwable> assertThrows(executable: () -> Unit): T {
  * ```
  * @see Assertions.assertThrows
  */
-inline fun <reified T : Throwable> assertThrows(message: String, executable: () -> Unit): T =
-    assertThrows({ message }, executable)
+inline fun <reified T : Throwable> assertThrows(message: String, executable: () -> Unit): T {
+    contract {
+        callsInPlace(executable, UNKNOWN)
+    }
```

Comment:
Is it possible to test for this `contract`?
As in: if it's removed, something should stop compiling or fail a test, right?
Probably the one in the related convo: https://github.com/junit-team/junit5/pull/3259#discussion_r1207297362

(This is probably true for all of them.)

Reply:
I haven't found a way to test `callsInPlace(..., UNKNOWN)` contracts.
I though having another function with a contract inside could work, but it seems, kotlin compiler doesn't check whether a function is really called in place in the functions it's passed to.
```kt
fun notWorkingTest(value: () -> Unit) {
    contract {
        callsInPlace(value, UNKNOWN)
    }

    assertThrows<AssertionError>(value) // works fine even when assertThrows doesn't have a contract
}
```